### ISSUE TITLE
T line path not being followed

### ISSUE THREAD

**Issue by MireiaY** [CityScope/CS_Simulation_GAMA#133]
Hi  Arnaud! I have been working on the cityScopableHousingChoice.gaml model lately. As I told you the last meeting, the t=0 scenario is based on the incentives we are applying and the amount of m2 that are being built in Volpe. Once that is set, the amount of people that would end up living and working in Volpe is calculated, based on their income profile, and they are assigned a commuting mode based on the results of the regression.

That said, I am now trying to use this model to show the dynamics of the surrounding throughout the day, so that it can be seen that at certain time of the day there are people (who do not live in Kendall) commuting from other parts of the region. They use the mobility mode they have been assigned (people using cars, bikes, bus access Kendall using road entry points) and those who are assigned the T use this. In order to reproduce the dynamics of the bus and the T I am using parts of the GameIt code, but I am having problems especially with the T (the buses are still not working perfectly well, but I will get there later on).

That is a little bit the context. The issue I would like to discuss is the following: if you play the simulation as it is now and you explore the Agent T2 (the orange line), you can see how it starts in T_stop(2) which is a boundary T entry point and heads towards T_stop(14) using T_line(8) as its path. Once T_stop(14) is reached, the target is T_stop(26) following the same T_line(8) path. T_line(8) finishes in the middle point between T_stop(14) and T_stop(26), so I guess that GAMA (you can check the variable graph_per_mobility_T for the orange line) should be able to jump from T_line(8) to T_line(9), so that T2 reaches T_stop(26). This is not happening, and once T2 reaches the point between T_stop(14) and T_stop(26) where T_line(8) finishes, it goes crazy.

This same situation is happening to T0 (red line). It starts in T_stop(0), boundary T entry point and heads towards T_stop(18)  and then to T_stop(5) using T_line(15). This T_line finishes between T_stop(5) and T_stop(19). In this case, once T_line(15) finishes, GAMA directly links T_line(16) to the path which is perfect, BUT when T_stop(19)  is reached, and T0 starts heading T_stop(25), the link between T_line(16) and T_line(17) is not made, so I end up having the same problem as in the orange line.

I hope I have explained myself well enough. The T line shapefile I am using is called "kendall_Tline.shp". I have checked if there are any dangling ends, so that GAMA does not detect the end of one T line section as the same location as the beginning of the next one, but apparently everything is fine. 

Thanks in advance!!

**Comment by agrignard** [comment]
Hi @MireiaY , I  am trying to work on your issue but to be totally honest I 'm not sure I understand very well yet the problem.

Would it be possible for you to make a dedicated experiment for this specific issue (in the smae model but creating another Experiment), where you only show (by playing with the layer species aspect etc) the element that is needed, so just the T_line, bus etc? I think it can help to debug it. You can maybe play also with a specific aspect for the T_Line to show there name etc. 

I am working on it right now and try to give you some answers if i found them ,-)

**Comment by agrignard** [comment]
This is the T Line where are speaking about right?
<img width="756" alt="Screenshot 2020-07-10 at 16 13 50" src="https://user-images.githubusercontent.com/3928502/87163960-66abbc00-c2c8-11ea-9c63-daab3f449279.png">

**Comment by MireiaY** [comment]
I have many more segments than you have. I will go ahead with this smaller experiment thing and will push the code again, see if it helps
![Tline](https://user-images.githubusercontent.com/58276088/87171179-57187d00-c2a0-11ea-8a8f-5907a9b0736e.png)

**Comment by agrignard** [comment]
ah sorry I was looking at the wrong file kendall_Tline I will look at the "big" one

**Comment by MireiaY** [comment]
(don't pay attention to the name difference, I will double check but this is the one I uploaded)

**Comment by MireiaY** [comment]
![image](https://user-images.githubusercontent.com/58276088/87171615-e9b91c00-c2a0-11ea-9d2d-71441abb0b57.png)

**Comment by MireiaY** [comment]
Yeap. I just changed the name so that github would not create another file in volpe. The last picture is what is on Github with the exact same name there

**Comment by MireiaY** [comment]
when pulling housingChoice do you just get 4 items in the .shp file of Tlines?

**Comment by agrignard** [comment]
Did you push everything?

**Comment by MireiaY** [comment]
Okay, I can tell that there is .qpj, .prj and the .cpg have not been updated on Github (they were pushed 14 days ago instead of 2)
. Let me work on that

**Comment by MireiaY** [comment]
Now that I think about it again, even if these 3 files are the same, the "column and row" info is on .dbf, the nature of the shape itself on the .shp and .shx the indexing, then this should not be a problem... I made a slight change in the kendall_Tline columns and indeed github only shows changes in the .dbf file... am I mistaken? Could you send me how you see the shape itself on QGIS?

**Comment by MireiaY** [comment]
Sth happened which is very surprising. I created the debugging experiments you commented and in order to debug I changed the geometry envelope from Roads (only 1kmx1km around kendall) to Roads_big (the slightly bigger scope I am trying to use). Now the T is functioning perfectly well.... Surprising, isn't it? Anyway, buses are still going crazy from time to time but I will have to simplify the roads for sure and then I will see if it is because of that. But what worries me is that you are getting different version of the files I am pushing. Could you at some point recheck if with this new push I have just made you are still getting the previous files (1kmx1km around Kendall, with only 4 T segments and super simple roads instead of the picture I attach here which is what you should get when you press  experiment "visual")
![image](https://user-images.githubusercontent.com/58276088/87206608-1b010e80-c2d8-11ea-815e-68bba12c1bd2.png)
.

**Comment by agrignard** [comment]
Ok so this is what I have after pulling your last version it seems that we are on the same page
<img width="984" alt="Screenshot 2020-07-11 at 00 41 46" src="https://user-images.githubusercontent.com/3928502/87208562-5ec33a80-c30f-11ea-9934-261c90b25479.png">

**Comment by MireiaY** [comment]
That is good news! Great! I will let you know if I am able to control or not the buses' craziness!

**Comment by LAAP** [comment]
Guys, congrats!!! This simulation is starting to look amazing!!!

**Comment by agrignard** [comment]
@MireiaY how is everything going? do you still have an issue here?

**Comment by MireiaY** [comment]
I did not work on this yesterday. I will try today and I will let you know. Thanks for asking

**Comment by MireiaY** [comment]
@agrignard The problem with the buses was indeed caused by the complexity of the roads. I simplified them heavily and these are working fine. I don't have a final road shp file yet though (when unifying the more detailed roads in Kendall and the less detailed one on the outskirts I am having overlapping problems). As for now, I defined two different species (major and minor roads) and I will work on this in the following days. The good news is that it seems like the code is ok  for now
